### PR TITLE
Clarifications to proposal 0005

### PR DESCRIPTION
**Pull request by joshuagl**
Very minor editorial changes to proposal 005 to make it (I think) easier to read:

* 088e29638957c05eb524a249be6685a20d826561 contains very minor clarity edits to the text from #9
* 14407613abc96c56540e3c5310e4cfe4cc208e50 adds my suggestion that the proposed conformance attestations _may_ come from auditor
* 01820eb8c3e841d80d94d790cca8b7ac2ba5483f populates the currently empty h4 "[Tier two: Third-party audit](https://github.com/slsa-framework/slsa-proposals/tree/2139c6f34cd75aa299af5a3d99f1fc5ce7de582c/0005#tier-two-third-party-audit)" with some content from later in the doc underneath a seemingly out of place H3 "[Build platform participation](https://github.com/slsa-framework/slsa-proposals/tree/2139c6f34cd75aa299af5a3d99f1fc5ce7de582c/0005#build-platform-participation-1)".
* 04738dfabb7c7dc3781c081a4bcb4d43f32c51f1 adds some minor clarity edits to the auditor participation section.

### CODE REVIEW

**Comment by joshuagl** [comment]
Apologies for not providing this feedback before #9 was merged, and thank you for all the great work in that PR!
I've submitted the changes in this PR as a linear series of patches in an attempt to make it easier to review where text is moved vs. where text is changed.

@kpk47 please let me know if you would like help setting up the conformance program (and please sign me up to be a maintainer of the build platform and auditor review repo(s)).

**Review by MarkLodato — APPROVED** [review]

**Review by kpk47 — APPROVED** [review]

**Comment by kpk47** [comment]
@joshuagl Thank you for the edits, and the offer to be a maintainer!

**Pull request merged**
